cmd/kubeadm/app/cmd/upgrade/apply.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	clientset "k8s.io/client-go/kubernetes"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	"k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/version"
+)
+
+// applyFlags holds the information about the flags that can be passed to apply
+type applyFlags struct {
+	nonInteractiveMode, force, dryRun bool
+	newK8sVersionStr                  string
+	newK8sVersion                     *version.Version
+	pullTimeout                       time.Duration
+	parent                            *cmdUpgradeFlags
+}
+
+// NewCmdApply returns the cobra command for `kubeadm upgrade apply`
+func NewCmdApply(parentFlags *cmdUpgradeFlags) *cobra.Command {
+	flags := &applyFlags{
+		parent:      parentFlags,
+		pullTimeout: 15 * time.Minute,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "apply",
+		Short: "Apply the upgrade",
+		Run: func(cmd *cobra.Command, args []string) {
+			kubeadmutil.CheckErr(ValidateFlags(flags))
+
+			// Default the flags
+			SetFlagDefaults(flags)
+
+			err := RunApply(flags)
+			kubeadmutil.CheckErr(err)
+		},
+	}
+
+	// Specify the valid flags specific for apply
+	cmd.Flags().BoolVarP(&flags.nonInteractiveMode, "yes", "y", flags.nonInteractiveMode, "Perform the upgrade and do not prompt for confirmation (non-interactive mode)")
+	cmd.Flags().BoolVarP(&flags.force, "force", "f", flags.force, "Force upgrading although some requirements might not be met. This also implies non-interactive mode")
```

Comment:
UI nitpick: If we're using a period in the middle of the string to end a sentence, it feels bare without a period at the end as well.